Services/CO.CDP.EntityVerification/Ppon/OrganisationRegisteredEventHandler.cs

```diff
@@ -12,7 +15,14 @@ public Task Handle(OrganisationRegistered @event)
 
         Persistence.Ppon newIdentifier = new() { PponId = pponId };
 
-        pponRepository.Save(newIdentifier);
+        using (TransactionScope txn = new TransactionScope())
+        {
+            pponRepository.Save(newIdentifier);
+            publisher.Publish(newIdentifier);
+
+            txn.Complete();
```

Comment:
Consider the following scenarios:

| Scenario  | In transaction | `r.Save(i)` | `p.Publish(i)` | Identifier saved | Message Published | Comment |
| ------------- | ------------- |------------- |------------- |------------- |------------- |------------- |
| Happy path (transaction)  | Yes  | 🟢   | 🟢  | 🟢  | 🟢  | |
| Happy path  | No   | 🟢   | 🟢  | 🟢  | 🟢  | |
| Repository fails (transaction)  | Yes  | 🔴 | 🟢 | 🔴 | 🟢 | Repository failure isn't triggered until the transaction is committed. The message is published even though the identifier isn't saved. |
| Repository fails | No  | 🔴  | 🟢  | 🔴  | 🔴  | |
| Publisher fails (transaction)  | Yes  | 🟢 | 🔴 |  🔴 | 🔴 | |
| Publisher fails | No  | 🟢  | 🔴  | 🟢  | 🔴  |  Identifier is saved, but the message isn't published. |


Is it a good idea to use a db transaction while the publisher isn't using the database?

It's all up to differences between "Repository fails (transaction)" and "Publisher fails", as all the other scenarios are as expected. 

Is it better to have a message published for identifier that wasn't saved, or to have an identifier saved and no message published?

As part of improving resiliency (separate story), we're planning to introduce the outbox pattern. That would make the repository and publisher can share the transaction. At this point in time, I think the transaction brings more problems.

Reply:
It's a shame we're not using a queue tech that supports transactions as this would solve the resiliency problem. I added the TransactionScope as I consider it best practice when going across multiple resources. If we could switch to ActiveMQ which does support transactions then we wouldn't need to implement the outbox pattern but I appreciate the spec says SQS.

Reply:
Let's remove the TransactionScope for now and we either re-add if we use e.g. ActiveMQ or we implement the outbox pattern.